crates/types/src/traits/state.rs

```diff
@@ -47,23 +46,23 @@ pub trait State:
     /// Time compatibility needed for reward collection
     type Time: ConsensusTime;
 
-    /// Returns true if and only if the provided block header is valid and can extend this state
-    fn validate_block(&self, block_header: &Self::BlockHeader, view_number: &Self::Time) -> bool;
-
-    /// Initialize the state.
-    fn initialize() -> Self;
-
-    /// Appends the given block header to this state, returning an new state
+    /// Check if the proposed block header is valid and apply it to the state if so.
+    ///
+    /// Returns the new state.
     ///
     /// # Errors
     ///
-    /// Should produce and error if appending this block header would lead to an invalid state
-    fn append(
+    /// If the block header is invalid or appending it would lead to an invalid state.
+    fn validate_and_apply_header(
         &self,
-        block_header: &Self::BlockHeader,
+        proposed_header: &Self::BlockHeader,
+        parent_header: &Self::BlockHeader,
         view_number: &Self::Time,
     ) -> Result<Self, Self::Error>;
 
+    /// Initialize the state.
+    fn initialize() -> Self;
```

Comment:
This is when we're forced to initialize from scratch because of catchup, right? I think we should at least get a `&Self::BlockHeader` in this case, since we should always have a leaf or proposal or something for a view where we're trying to figure out the state. Right?

We need this in the sequencer so we can use [`from_commitment`](https://jellyfish.docs.espressosys.com/jf_primitives/merkle_tree/trait.ForgetableMerkleTreeScheme.html#tymethod.from_commitment) to build a completely sparse merkle tree from the merkle commitment contained in the header.

Reply:
Oh, I only moved but didn't modify this function. Actually, let me delete this function for now since catchup is out of the scope of this PR. When I work on the state building for catchup, I'll add a function with additional parameter(s) as you commented here.